core/src/test/java/org/elasticsearch/search/suggest/SuggestSearchIT.java

```diff
@@ -1108,6 +1119,231 @@ public void testSuggestWithManyCandidates() throws InterruptedException, Executi
         // assertThat(total, lessThan(1000L)); // Takes many seconds without fix - just for debugging
     }
 
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singleton(DummyTemplatePlugin.class);
+    }
+
+    public static class DummyTemplatePlugin extends Plugin implements ScriptPlugin {
+        @Override
+        public ScriptEngineService getScriptEngineService(Settings settings) {
+            return new DummyTemplateScriptEngine();
+        }
+    }
+
+    public static class DummyTemplateScriptEngine implements ScriptEngineService {
+
+        public static final String NAME = "mustache";
+
+        @Override
+        public void close() throws IOException {
+        }
+
+        @Override
+        public String getType() {
+            return NAME;
+        }
+
+        @Override
+        public String getExtension() {
+            return NAME;
+        }
+
+        @Override
+        public Object compile(String scriptName, String scriptSource, Map<String, String> params) {
+            return scriptSource;
+        }
+
+        @Override
+        public ExecutableScript executable(CompiledScript compiledScript, Map<String, Object> params) {
+            String script = (String) compiledScript.compiled();
+            for (Entry<String, Object> entry : params.entrySet()) {
+                script = script.replace("{{" + entry.getKey() + "}}", String.valueOf(entry.getValue()));
+            }
+            String result = script;
+            return new ExecutableScript() {
+                @Override
+                public void setNextVar(String name, Object value) {
+                }
```

Comment:
I think it'd be nice to throw an UnsupportedOperationException here just to catch any weird usage quickly.